extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1321,6 +1356,74 @@ export class Project implements ISqlProject {
 		}
 	}
 
+	/**
+	 * Adds a value to project property, separated by semicolon if
+	 * one or more values are already present in project property
+	 * @param propertyName Name of property
+	 * @param valueToAdd Value to be added to project property, values containing semicolon are not supported
+	 */
+	private async addValueToCollectionProjectProperty(propertyName: string, valueToAdd: string) {
+		if (valueToAdd.includes(';')) {
+			return;
+		}
+
+		let propertyValue = this.evaluateProjectPropertyValue(propertyName);
+		if (propertyValue === undefined) {
+			propertyValue = valueToAdd;
+		} else {
+			if ((propertyValue.length > valueToAdd.length
+				&& (propertyValue.startsWith(`${valueToAdd};`)
+					|| propertyValue.endsWith(`;${valueToAdd}`)
+					|| propertyValue.indexOf(`;${valueToAdd};`) > 0))
+				|| (propertyValue.length === valueToAdd.length
+					&& propertyValue === valueToAdd)) {
+				return;
+			}
+
+			propertyValue = `${propertyValue};${valueToAdd}`;
+		}
+
+		await this.setProjectPropertyValue(propertyName, propertyValue);
+	}
+
+	/**
+	 * Removes value from project property where values are semicolon separated
+	 * @param propertyName Name of property
+	 * @param valueToRemove Value to remove from the project property
+	 */
+	private async removeValueFromCollectionProjectProperty(propertyName: string, valueToRemove: string) {
+		const propertyValue = this.evaluateProjectPropertyValue(propertyName);
+		if (propertyValue === undefined
+			|| propertyValue.length < valueToRemove.length
+			|| valueToRemove.includes(';')) {
+			return;
+		}
+
+		if (propertyValue.length > valueToRemove.length) {
+			let propertyValueToSet = propertyValue;
+			let valueToRemovePosition: number;
+
+			if (propertyValue.startsWith(`${valueToRemove};`)) {
+				propertyValueToSet = propertyValue.substring(valueToRemove.length + 1);
+			} else if (propertyValue.endsWith(`;${valueToRemove}`)) {
+				propertyValueToSet = propertyValue.substring(0, propertyValue.length - valueToRemove.length - 1);
+			} else if ((valueToRemovePosition = propertyValue.indexOf(`;${valueToRemove};`)) > 0) {
+				propertyValueToSet = propertyValue.substring(0, valueToRemovePosition + 1)
+					+ propertyValue.substring(valueToRemovePosition + valueToRemove.length + 2);
+			} else {
+				return;
+			}
+
+			await this.setProjectPropertyValue(
+				propertyName,
+				propertyValueToSet);
+		} else if (propertyValue.length === valueToRemove.length
```

Comment:
If property is set to `;;;MySource;;` and someone wants to remove `MySource` it will be covered under the case
`else if ((valueToRemovePosition = propertyValue.indexOf(`;${valueToRemove};`)) >= 0)`
However, it will remove one of the `;` because that's typically what we would want to do.

Reply:
Right: `;;;MySource;;` should collapse to four semicolons, not five.